### PR TITLE
PaletteEdit: Use `ItemGroup` and `Item`, and avoid custom styles

### PR DESCRIPTION
**Pull request by tyxla**
## What?
This PR updates `PaletteEdit` to make use of `Item` instead of implementing custom styles.

## Why?
`PaletteEdit` currently uses `ItemGroup` in an odd way, and defines its own `Item` with a bunch of custom styles. For better consistency, when using `ItemGroup` we should be using `Item` components.

## How?
- Updating `PaletteEdit` to use `ItemGroup` with `Item`, instead of building its own custom `PaletteItem`
- Cleaning up a bunch of redundant styles related to the removed custom `PaletteEdit` palette item.

Since this uses the `Item` built-in spacing, this introduces some visual changes, so I could use a look from @WordPress/gutenberg-design. I'm happy to address any suggestions while here, but they'll need to be done inside `Item`, and AFAIK, we're aiming to work on this separately (see https://github.com/WordPress/gutenberg/issues/64445). 

## Testing Instructions
* Spin up a Storybook instance and test `PaletteEdit`:
  * Click the vertical ellipsis
  * Click "Show details"
  * Add/remove colors
  * Verify borders and spacing makes sense across different viewports. 
* Test palette editing in the site editor:
  * Go to the site editor and open the Styles sidebar
  * Click "Colors" and then the active Palette.
  * Add/edit the custom palette and verify it looks good and works well.

### Testing Instructions for Keyboard
Same

## Screenshots or screencast 

Storybook, `PaletteEdit`, default story, editing palette:

|Before|After|
|---|---|
|Screenshot 2024-10-16 at 12 47 17|Screenshot 2024-10-16 at 12 47 34|

Site Editor, global styles, colors, editing palette:

|Before|After|
|---|---|
|Screenshot 2024-10-16 at 12 42 58|Screenshot 2024-10-16 at 12 42 12|

### CODE REVIEW

File: packages/components/CHANGELOG.md [hunk 1]
```diff
@@ -19,6 +19,8 @@
 
 -   `Modal`: Modal dialog small improvement for elementShouldBeHidden ([#65941](https://github.com/WordPress/gutenberg/pull/65941)).
 -   `Tabs`: revamped vertical orientation styles ([#65387](https://github.com/WordPress/gutenberg/pull/65387)).
+-   `ItemGroup`: default `isBordered` and `isSeparated` to `true` ([#66164](https://github.com/WordPress/gutenberg/pull/66164)).
```
Comment (ciampo) [inline packages/components/CHANGELOG.md]: I'm not sure I understand why is this change needed — couldn't we just set the two props to `true` in `PaletteEdit`?
(also, technically this is a breaking change and we'd normally try to avoid that as much as possible)

Comment (tyxla) [reply packages/components/CHANGELOG.md]: As described in the PR, I'm fine with a breaking change for a few reasons:
- The sole usage of this component is in `PaletteEdit`
- The default non-bordered behavior of the component is awkward
- The component is experimental, so it's perfectly OK to set better defaults, even through a breaking change. 

Related to the CHANGELOG entry, if this is a breaking change to an experimental API, do we treat it as a breaking change? It wasn't an official public API, after all. Happy to move it to "Breaking Changes", but to me it makes sense if that's reserved to changes to components that aren't private, experimental or unstable.

Comment (ciampo) [reply packages/components/CHANGELOG.md]: > The sole usage of this component is in `PaletteEdit`

That's definitely a point to keep in mind when judging the impact of these changes — although I haven't looked on third-party usage.

> The default non-bordered behavior of the component is awkward

That's absolutely true. In general, there's a lot of [improvement work on `ItemGroup`](https://github.com/wordPress/gutenberg/issues/64445) on the horizon

> The component is experimental, so it's perfectly OK to set better defaults, even through a breaking change.

While the component is marked as experimental, it is _de facto_ considered stable by some folks given that it was included in a WordPress release — hence why I questioned this change in the first place.

Having said that, we can definitely introduce this change if we think it improves the component — we should provide a dev note to let folks know (with an example of how to revert to previous styling, though).

> Related to the CHANGELOG entry, if this is a breaking change to an experimental API, do we treat it as a breaking change? It wasn't an official public API, after all. Happy to move it to "Breaking Changes", but to me it makes sense if that's reserved to changes to components that aren't private, experimental or unstable.

I'm tempted to mark it as a Breaking Change, since I believe that recently we've been treating the "Experimental" section of the changelog as referred to the truly experimental (ie. exposed via private APIs) components. But I don't have a strong opinion — @mirka , do you have any thoughts?

Comment (tyxla) [reply packages/components/CHANGELOG.md]: > That's definitely a point to keep in mind when judging the impact of these changes — although I haven't looked on third-party usage.

We should be good: https://wpdirectory.net/search/01JAAG2CQV99VX0PZX2PER2BFA

> That's absolutely true. In general, there's a lot of improvement work on ItemGroup on the horizon

Yep, thanks for confirming, I actually did link that in the PR description already. 

> While the component is marked as experimental, it is de facto considered stable by some folks given that it was included in a WordPress release — hence why I questioned this change in the first place.

Given that the name hints it is experimental, and docs are clear, I think the limitations here are self-imposed, and we need to be more flexible. Given that I don't see any other usage of that component, I'm confident to make this change at this time.

> Having said that, we can definitely introduce this change if we think it improves the component — we should provide a dev note to let folks know (with an example of how to revert to previous styling, though).

Good point, happy to provide a dev note if we end up landing it.

> I'm tempted to mark it as a Breaking Change, since I believe that recently we've been treating the "Experimental" section of the changelog as referred to the truly experimental (ie. exposed via private APIs) components. But I don't have a strong opinion — @mirka , do you have any thoughts?

Happy to be flexible with whatever the consensus is - let me know.

**Comment by mirka** [comment]
In general I've been treating the `__experimental` components to be de facto stable at this point, which doesn't mean we _can't_ make breaking changes, but that the decision making around it should basically be the same as if it were a stable component. Is it a "self-imposed limitation"? Yes 😂 But also kind of easier to reason about, since the waters are murky to begin with.

> The sole usage of this component is in `PaletteEdit`

I'm not sure I understand this correctly, because I'm seeing a good number of [`ItemGroup` usage](https://github.com/search?q=repo%3AWordPress%2Fgutenberg+%22%3CItemGroup%22+-path%3A%2F%5Epackages%5C%2Fcomponents%5C%2Fsrc%5C%2F%2F&type=code) in the app that don't enable the bordered and separated props.

I do lean toward keeping the default-flipping a separate discussion, especially since we are already planning to overhaul the component anyway. We can probably also do more to improve the situation through documentation — for example making the bordered & separated example more prominent. Or even making the default story have the two props enabled. We usually showcase the default state, but in this case I can see the advantages.

The other changes in this PR sound good though 👍

**Comment by ciampo** [comment]
> Or even making the default story have the two props enabled. We usually showcase the default state, but in this case I can see the advantages.

This would be my preferred short-term solution.

With the long-term solution being that we rework the component to follow [the new specs](https://github.com/wordPress/gutenberg/issues/64445) — and that's where applying breaking changes (or introducing a new version altogether) would make more sense.

**Comment by tyxla** [comment]
Thanks for the feedback @mirka @ciampo 🙌 

I've addressed it, keeping this PR solely for the `PaletteEdit` changes. 

I've also opened #66191 to improve the `ItemGroup` stories without actually touching any defaults.

This is ready for another look 👀

**Review by ciampo — APPROVED** [review]
> Since this uses the `Item` built-in spacing, this introduces some visual changes, so I could use a look from @WordPress/gutenberg-design. I'm happy to address any suggestions while here, but they'll need to be done inside `Item`, and AFAIK, we're aiming to work on this separately (see #64445).

I noticed that originally `PaletteEdit` was using `Item`, but was changed in https://github.com/WordPress/gutenberg/pull/38035 to a div + custom styles to fix a double-border bug (cc @jorgefilipecosta ) — although these hover + selected styles don't seem to be there anymore (probably as the result of https://github.com/WordPress/gutenberg/pull/62753)? (this can be tested especially when setting the `canOnlyChangeValues` to `true`).

Not sure if we want to make any tweaks there? But since this seem to be same on `trunk`, any changes could happen in a follow-up.

Apart from that, changes LGTM 🚀

**Comment by tyxla** [comment]
> Not sure if we want to make any tweaks there? But since this seem to be same on `trunk`, any changes could happen in a follow-up.

Sounds good; I'm happy to make any changes in a follow-up.
